code/game/machinery/computer/atmos_control.dm

```diff
@@ -0,0 +1,589 @@
+#define ACA_SCREEN_DETAILSVIEW 1
+#define ACA_SCREEN_ADMINPANEL 2
+
+var/global/list/atmos_controllers = list()
+/obj/item/weapon/circuitboard/atmoscontrol
+	name = "\improper Central Atmospherics Computer Circuitboard"
+	build_path = /obj/machinery/computer/atmoscontrol
+
+/obj/machinery/computer/atmoscontrol
+	name = "\improper Central Atmospherics Computer"
+	icon = 'icons/obj/computer.dmi'
+	icon_state = "tank"
+	density = 1
+	anchored = 1.0
+	circuit = "/obj/item/weapon/circuitboard/atmoscontrol"
+	var/obj/machinery/alarm/current
+	var/list/filter=null
+	var/obj/item/weapon/card/id/log_in_id = null //the ID that's currently logged in
+	var/screen = ACA_SCREEN_DETAILSVIEW //the current screen in the UI
+	var/datum/airalarm_preset/selected_preset = null //stores the preset settings while they're being edited
+
+	light_color = LIGHT_COLOR_CYAN
+
+/obj/machinery/computer/atmoscontrol/New()
+	..()
+	atmos_controllers += src
+/obj/machinery/computer/atmoscontrol/Destroy()
+	atmos_controllers -= src
+	..()
+
+/obj/machinery/computer/atmoscontrol/xeno
+	name = "\improper Xenobiology Atmospherics Computer"
+	filter=list(
+		/area/science/xenobiology/specimen_1,
+		/area/science/xenobiology/specimen_2,
+		/area/science/xenobiology/specimen_3,
+		/area/science/xenobiology/specimen_4,
+		/area/science/xenobiology/specimen_5,
+		/area/science/xenobiology/specimen_6
+	)
+
+/obj/machinery/computer/atmoscontrol/gas_chamber
+	name = "\improper Gas Chamber Atmospherics Computer"
+	filter=list(
+		/area/security/gas_chamber
+	)
+
+/obj/machinery/computer/atmoscontrol/attack_ai(var/mob/user as mob)
+	src.add_hiddenprint(user)
+	return interact(user)
+
+/obj/machinery/computer/atmoscontrol/attack_paw(var/mob/user as mob)
+	return interact(user)
+
+/obj/machinery/computer/atmoscontrol/attack_hand(mob/user)
+	if(..())
+		return
+	return interact(user)
+
+/obj/machinery/computer/atmoscontrol/interact(mob/user)
+	return ui_interact(user)
+
+
+/obj/machinery/computer/atmoscontrol/attackby(var/obj/item/I as obj, var/mob/user as mob)
+	if(istype(I, /obj/item/weapon/card/emag))
+		emag(user)
+	return ..()
+
+/obj/machinery/computer/atmoscontrol/emag(var/obj/item/emag as obj, var/mob/user as mob)
+	if(!emagged)
+		emagged = 1
+		new/obj/effect/effect/sparks(get_turf(src))
```

Comment:
use a spark system, if my memory serves the sparks will never get deleted if you do it this way

Reply:
Funny enough, I was about to report this today. Someone should really fix that.

Reply:
`spark(src, 1, FALSE)` gonna work?